crates/ruff/src/rules/pyflakes/rules/f_string_missing_placeholders.rs

```diff
@@ -47,31 +47,49 @@ impl AlwaysAutofixableViolation for FStringMissingPlaceholders {
     }
 }
 
-/// Find f-strings that don't contain any formatted values in an [`FString`].
+/// Return an iterator containing a two-element tuple for each f-string in the
+/// given [`ExprFString`] expression. The first element of the tuple is the
+/// f-string prefix range, and the second element is the entire f-string range.
+///
+/// For example,
+///
+/// ```python
+/// f"example"
+/// ```
+///
+/// would return `[(0..1, 0..10)]`.
+///
+/// This function assumes that the given expression is an [`ExprFString`]
+/// without any placeholder expressions.
+///
+/// [`ExprFString`]: `ruff_python_ast::ExprFString`
 fn find_useless_f_strings<'a>(
     expr: &'a Expr,
     locator: &'a Locator,
     source_type: PySourceType,
 ) -> impl Iterator<Item = (TextRange, TextRange)> + 'a {
     let contents = locator.slice(expr);
+    let mut current_f_string_start = TextSize::default();
     lexer::lex_starts_at(contents, source_type.as_mode(), expr.start())
         .flatten()
-        .filter_map(|(tok, range)| match tok {
-            Tok::String {
-                kind: StringKind::FString | StringKind::RawFString,
-                ..
-            } => {
-                let first_char = locator.slice(TextRange::at(range.start(), TextSize::from(1)));
+        .filter_map(move |(tok, range)| match tok {
+            Tok::FStringStart => {
+                current_f_string_start = range.start();
```

Comment:
How does this work with multiple nested fstrings? 

```python
f"{f"{a + f"{b}"}"}"
```

I would expect that `f"{b}"` overrides the start position of `f"{a + f"{b}"}"`

Reply:
To give some context, the autofix for the rule `F541` is to remove the `f` prefix from the f-string _without_ any placeholders. The placeholder checks are done at an expression level by checking if there's any `FormattedValue`. The function you're referring to is to get the `f` prefix range and the f-string range. At this point there can't be any nested f-strings.

Reply:
I think the function name is a bit confusing which I'll rename. I've also added docs to the function.

Reply:
Hmm, but the documentation mentions that this function returns all nested f-string ranges, meaning there need to be placeholders? It's also unclear to me where the filtering out of f-strings without placeholders would happen. 

So I think we need to update the documentation to match its actual behavior (or it's too early in the morning and I shouldn't be reviewing PR's :sweat_smile: )

Reply:
> Hmm, but the documentation mentions that this function returns all nested f-string ranges, meaning there need to be placeholders?

I've updated the documentation but I'm curious as to where was "nested f-string ranges" mentioned? 😅